lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -205,19 +207,19 @@ def __get_k8s_job_spec_template(self, ajs):
         k8s_spec_template = {
             "metadata": {
                 "labels": {
-                    "app.kubernetes.io/name": ajs.job_wrapper.tool.old_id,
+                    "app.kubernetes.io/name": self.LABEL_REGEX.sub("_", ajs.job_wrapper.tool.old_id),
                     "app.kubernetes.io/instance": self.__produce_k8s_job_prefix(),
-                    "app.kubernetes.io/version": ajs.job_wrapper.tool.version,
+                    "app.kubernetes.io/version": self.LABEL_REGEX.sub("_", ajs.job_wrapper.tool.version),
                     "app.kubernetes.io/component": "tool",
                     "app.kubernetes.io/part-of": "galaxy",
                     "app.kubernetes.io/managed-by": "galaxy",
-                    "app.galaxyproject.org/job_id": ajs.job_wrapper.get_id_tag(),
-                    "app.galaxyproject.org/instance": self._galaxy_instance_id or "",
-                    "app.galaxyproject.org/handler": self.app.config.server_name,
-                    "app.galaxyproject.org/destination": ajs.job_wrapper.job_destination.id,
+                    "app.galaxyproject.org/job_id": self.LABEL_REGEX.sub("_", ajs.job_wrapper.get_id_tag()),
+                    "app.galaxyproject.org/handler": self.LABEL_REGEX.sub("_", self.app.config.server_name),
+                    "app.galaxyproject.org/destination": self.LABEL_REGEX.sub(
+                        "_", str(ajs.job_wrapper.job_destination.id))
```

Comment:
@mvdbeek Something unexpected just happened - the following error came up, presumably because it was necessary to cast job_destination.id to str.
```
galaxy.jobs.runners ERROR 2020-10-22 01:05:46,092 [p:15,w:1,m:0] [KubernetesRunner.work_thread-1] (3) Unhandled exception calling queue_job
Traceback (most recent call last):
File "/galaxy/server/lib/galaxy/jobs/runners/__init__.py", line 137, in run_next
method(arg)
File "/galaxy/server/lib/galaxy/jobs/runners/kubernetes.py", line 131, in queue_job
self.__get_k8s_job_spec(ajs)
File "/galaxy/server/lib/galaxy/jobs/runners/kubernetes.py", line 196, in __get_k8s_job_spec
k8s_job_spec = {"template": self.__get_k8s_job_spec_template(ajs),
File "/galaxy/server/lib/galaxy/jobs/runners/kubernetes.py", line 218, in __get_k8s_job_spec_template
"app.galaxyproject.org/destination": self.LABEL_REGEX.sub("_", ajs.job_wrapper.job_destination.id)
TypeError: expected string or bytes-like object
galaxy.jobs.runners.kubernetes ERROR 2020-10-22 01:05:46,093 [p:15,w:1,m:0] [KubernetesRunner.work_thread-1] (3/None) User killed running job, but error encountered during termination: job name must not be empty
Traceback (most recent call last):
File "/galaxy/server/lib/galaxy/jobs/runners/kubernetes.py", line 515, in stop_job
job_to_delete = find_job_object_by_name(self._pykube_api, job.get_job_runner_external_id(), self.runner_params['k8s_namespace'])
File "/galaxy/server/lib/galaxy/jobs/runners/util/pykube_util.py", line 63, in find_job_object_by_name
raise ValueError("job name must not be empty")
ValueError: job name must not be empty
```
The unexpected thing is that the tests passed.

Reply:
perhaps job_destination.id is not what was intended. I wanted the destination id as specified in the job_conf. That should always be a string.

Reply:
Are you using a dynamic destination that maybe doesn't have an id ? If you can reproduce this, can you check if the destination id was `None` or an integer ? I think that would explain why the tests pass. I think you may want to cast tool.version to string. Yaml tools can specify integers or floats as version. That they are strings is an artefact of parsing xml (and if you use a job config yaml file with integer destination ids that might be another explanation why the test didn't fail).

Reply:
Thanks, that makes sense. Will update PR.